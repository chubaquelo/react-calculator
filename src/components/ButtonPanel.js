import React from 'react'
import Button from './Button'

export default function ButtonPanel() {
  return (
    <div>
      <Button />
    </div>
  )
}
