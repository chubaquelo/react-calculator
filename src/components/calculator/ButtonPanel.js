import React from 'react';
import PropTypes from 'prop-types';
import Button from './Button';

export default function ButtonPanel({ updateCalc, handleEqual }) {
  return (
    <>
      <div className="w-full border-white border-4 border-b-0 flex flex-wrap">
        <Button
          btnName="AC"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
        <Button
          btnName="+/-"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
        <Button
          btnName="%"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
        <Button
          btnName="/"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
      </div>
      <div className="w-full border-white border-4 border-b-0 flex flex-wrap">
        <Button btnName="7" updateCalc={updateCalc} addClass="text-3xl" />
        <Button btnName="8" updateCalc={updateCalc} addClass="text-3xl" />
        <Button btnName="9" updateCalc={updateCalc} addClass="text-3xl" />
        <Button
          btnName="x"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
      </div>
      <div className="w-full border-white border-4 border-b-0 flex flex-wrap">
        <Button btnName="4" updateCalc={updateCalc} addClass="text-3xl" />
        <Button btnName="5" updateCalc={updateCalc} addClass="text-3xl" />
        <Button btnName="6" updateCalc={updateCalc} addClass="text-3xl" />
        <Button
          btnName="-"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
      </div>
      <div className="w-full border-white border-4 border-b-0 flex flex-wrap">
        <Button btnName="1" updateCalc={updateCalc} addClass="text-3xl" />
        <Button btnName="2" updateCalc={updateCalc} addClass="text-3xl" />
        <Button btnName="3" updateCalc={updateCalc} addClass="text-3xl" />
        <Button
          btnName="+"
          updateCalc={updateCalc}
          addClass="bg-blue-300 hover:bg-blue-600"
        />
      </div>
      <div className="w-full border-white border-4 flex flex-wrap">
        <Button btnName="0" updateCalc={updateCalc} addClass="w-2/4 text-3xl" />
        <Button btnName="." updateCalc={updateCalc} addClass="text-3xl" />
        <Button
          btnName="="
          updateCalc={handleEqual}
          addClass="text-3xl bg-blue-400 hover:bg-blue-900"
        />
      </div>
    </>
  );
}

ButtonPanel.propTypes = {
  updateCalc: PropTypes.func,
  handleEqual: PropTypes.func,
};

ButtonPanel.defaultProps = {
  updateCalc: 0,
  handleEqual: 0,
};
