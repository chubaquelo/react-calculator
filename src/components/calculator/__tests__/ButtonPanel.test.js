import React from 'react';
import renderer from 'react-test-renderer';
import ButtonPanel from '../ButtonPanel';

describe.only('Test ButtonPanel Snapshot', () => {
  it('matches ButtonPanel snapshot', () => {
    const forTest = renderer.create(<ButtonPanel />).toJSON();
    expect(forTest).toMatchSnapshot();
  });
});
