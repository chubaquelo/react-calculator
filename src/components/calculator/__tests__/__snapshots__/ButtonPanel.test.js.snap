// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ButtonPanel Snapshot matches ButtonPanel snapshot 1`] = `
Array [
  <div
    className="w-full border-white border-4 border-b-0 flex flex-wrap"
  >
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
  </div>,
  <div
    className="w-full border-white border-4 border-b-0 flex flex-wrap"
  >
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
  </div>,
  <div
    className="w-full border-white border-4 border-b-0 flex flex-wrap"
  >
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
  </div>,
  <div
    className="w-full border-white border-4 border-b-0 flex flex-wrap"
  >
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl bg-blue-300 hover:bg-blue-600"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
  </div>,
  <div
    className="w-full border-white border-4 flex flex-wrap"
  >
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl w-2/4 text-3xl"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="border-white select-none border-2 w-1/4 h-12 flex justify-center items-center bg-red-500 text-white font-black cursor-pointer hover:bg-red-900 text-xl text-3xl bg-blue-400 hover:bg-blue-900"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>,
]
`;
